testnet/stacks-node/src/nakamoto_node/sign_coordinator.rs

```diff
@@ -250,7 +254,33 @@ impl SignCoordinator {
             ..Default::default()
         };
 
-        let mut coordinator: FireCoordinator<Aggregator> = FireCoordinator::new(coord_config);
+        let total_weight =
+            reward_set_signers
+                .iter()
+                .cloned()
+                .map(|s| s.weight)
+                .fold(0, |w, acc| {
+                    acc.checked_add(w)
+                        .expect("FATAL: Total signer weight > u32::MAX")
+                });
```

Comment:
I think this would be nice to implement as an instance method in `RewardSet`, i.e.:

```rust
pub fn total_signing_weight(&self) -> Result<u32, ChainstateError> {
       let Some(ref reward_set_signers) = self.signers else {
            error!("Could not initialize WSTS coordinator for reward set without signer");
            return Err(ChainstateError::NoRegisteredSigners(0));
       };
       reward_set_signers
                .iter()
                .fold(0, |s, acc| {
                    acc.checked_add(s.weight)
                        .expect("FATAL: Total signer weight > u32::MAX")
                })
}
```

Reply:
👍🏼 , added and am now using it in `SignCoordinator::new` and `verify_signer_signature`